toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/utils/ConfigTesterHelper.java

```diff
@@ -0,0 +1,220 @@
+											  boolean runTryValue,
+											  boolean fileOrDirectory) {
+		String fileName = directory.split(FILE_DIRECTORY_SEPARATOR)[directory.split(FILE_DIRECTORY_SEPARATOR).length - 1];
+		
+		boolean isKo = true;
+		if (fileOrDirectory) {
+			if (testFileDirectory(directory, runTryValue, fileName)) {
+				isKo = false;
+			}
+		} else {
+			if (testDirectory(directory, runTryValue, fileName)) {
+				isKo = false;
+			}
+		}
+		return !isKo;
+	}
+
+	public static boolean testDirectory(String directory, boolean runTryValue, String fileName) {
+		File myFile = new File(directory);
+		if (myFile.exists()) {
+			if (myFile.isDirectory()) {
+				return true;
+			} else {
+				if (runTryValue) {
+					NotificationManager.showMessage("You did not select a directory.").showNotification();
+				}
+				return false;
+			}
+		} else {
+			if (runTryValue) {
+				NotificationManager.showMessage("The directory : " + directory.split(fileName)[0] + " does not exist !")
+						.showNotification();
+			}
+			return false;
+		}
+	}
+
+	public static boolean testFileDirectory(String directory, boolean runTryValue, String fileName) {
+		File myFile = new File(directory);
+		if (myFile.exists()) {
+			if (myFile.isFile()) {
+				return true;
+			} else {
+				if (runTryValue) {
+					NotificationManager.showMessage("You did not select a file.").showNotification();
+				}
+				return false;
+			}
+		} else {
+			if (runTryValue) {
+				NotificationManager
+						.showMessage(fileName + " does not exist in the directory : " + directory.split(fileName)[0])
+						.showNotification();
+			}
+			return false;
+		}
+	}
+
+	public static boolean testWebAppUrl(String url, boolean runTryValue) throws IOException {
+		return testWebAppUrl(url, runTryValue, null, null, null, null);					
+	}
+
+	public static boolean testWebAppUrl(String url, boolean runTryValue, String proxyAdress, String proxyPort,
+			String proxyUserName, String proxyUserPswd) throws IOException {
+
+		boolean isKo = true;
+		if (url.contains(" ")) {
+			if (runTryValue) {
+				NotificationManager.showMessage(url + " has spaces in its name.").showNotification();
+			}
+		}
+
+		if (getStatus(url,proxyAdress, proxyPort, proxyUserName, proxyUserPswd)) {
+			isKo = false;
+		} else if (runTryValue) {
+			NotificationManager.showMessage(url + " does not answer.").showNotification();
+		}
+		return !isKo;
+	}
+
+	public static boolean getStatus(String url, String proxyAdress, String proxyPort, String proxyUserName,
+			String proxyUserPswd) throws IOException {
+
+	        boolean result = false;
+	        Proxy proxy = null;
+	        if(proxyAdress != null && proxyPort != null) {
+				InetSocketAddress proxyInet = new InetSocketAddress(proxyAdress,Integer.parseInt(proxyPort));
+				proxy = new Proxy(Proxy.Type.HTTP, proxyInet);
+	        }
+	        
+	        Authenticator.setDefault (new Authenticator() {
+	            protected PasswordAuthentication getPasswordAuthentication() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1161)